src/l500/ac-trigger.cpp

```diff
@@ -681,12 +682,27 @@ namespace ivcam2 {
 
                     auto df = _sf.get_depth_frame();
                     auto irf = _sf.get_infrared_frame();
-                    depth_to_rgb_calibration algo( df, irf, _cf, _pcf, cal_info, cal_regs );
+
+                    depth_to_rgb_calibration algo(df, irf, _cf, _pcf, cal_info, cal_regs, 
+                    [&]() 
+                    {
+                        if (!is_processing())
+                        {
+                            AC_LOG( DEBUG , "The device has been disconnected during Camera Accuracy Health process");
+                            throw librealsense::camera_disconnected_exception("Stopping CAH algorithm");
+                        }
+                    });
+
                     _from_profile = algo.get_from_profile();
                     _to_profile = algo.get_to_profile();
 
-                    auto status = algo.optimize(
-                        [this]( rs2_calibration_status status ) { call_back( status ); } );
+                    rs2_calibration_status status(RS2_CALIBRATION_FAILED); // assume fail until gets a success status.
+
+                    if (is_processing()) // check if the device still exists
```

Comment:
How is this different than is_active() check in line 709?

Reply:
is_active is set  when the we want algo to start and set to false when alog finished or stop function is called.
is_processing is set when the algo actually starts and set to false on the Dtor (stop is not called is this process)

Reply:
Check that is_active() is always true is is_processing() is true: stop() should set is_processing() to false!

Reply:
Done